src/content/docs/apis/nerdgraph/examples/nerdgraph-streaming-export.mdx

```diff
@@ -407,6 +407,12 @@ You can enable compression using the `payloadCompression` field under `ruleParam
 
 GZIP is the only compression format currently available, though we may choose to make more formats available in the future.
 
+Notably, when compression is enabled on a pre-existing export rule, it's possible for the next batch to contain both compressed and decompressed data. This has been observed in AWS, and is caused by batching within Kinesis Firehose, though it may be present in Azure as well. This can be avoided by temporarily disabling the export rule before enabling compression, or by creating a new, separate Kinesis Firehose stream for the compressed data to flow through. If this does happen to you, it is possible to view the compressed part of the data by:
+1. Manually downloading the object.
+2. Separating it into two separate files by copy-pasting the compressed data into a new file.
+3. Decompressing the new, compressed-only data file.
+4. If desired, you can then choose to re-upload the two separate files back to S3 (or whatever other service you're using), and delete the old file.
+
```

Comment:
```suggestion
Notably, when compression is enabled on a pre-existing export rule, it's possible for the next batch to contain both compressed and decompressed data. This has been observed in AWS, and is caused by batching within Kinesis Firehose, though it may be present in Azure as well. This can be avoided by temporarily disabling the export rule before enabling compression, or by creating a new, separate Kinesis Firehose stream for the compressed data to flow through. If this does happen to you, it is possible to view the compressed part of the data by:
1. Manually downloading the object.
2. Separating it into two separate files by copy-pasting the compressed data into a new file.
3. Decompressing the new, compressed-only data file.

If desired, you can then re-upload the two separate files back to S3 (or whatever other service you're using) and delete the old file.

```

Reply:
@ecchilds How does this revision look?